src/library_syscall.js

```diff
@@ -1709,6 +1709,9 @@ for (var x in SyscallsLibrary) {
   var canThrow = SyscallsLibrary[x + '__nothrow'] !== true;
   delete SyscallsLibrary[x + '__nothrow'];
   var handler = '';
+#if SYSCALLS_REQUIRE_FILESYSTEM == 0 && MINIMAL_RUNTIME
```

Comment:
Hmm, if e.g. one attempts to open a file that does not exist, or e.g. [delete a directory that does not exist](https://github.com/emscripten-core/emscripten/blob/master/src/library_memfs.js#L226), the filesystem layer will throw an exception.

These exceptions need to be caught here in the syscalls so that they will not abort execution by unwinding through the wasm/C code. Even if building with MINIMAL_RUNTIME and without filesystem, we'd like to avoid these exceptions from propagating through, but instead attempting a file operation that fails would return a failure to C side (a null FILE handle for example?

Reply:
> we'd like to avoid these exceptions from propagating through

Well, the call to `abort()` in the `catch(){}` statement will still cause the exceptions to propagate through, since it is throwing the caught exception.

So, these changes are just meant to prevent such redundant try/catches that does yield the exception propagating by invoking abort()

Reply:
For example, see this code:

```js
function abort(what) {
  throw what;
}

function _fd_fdstat_get(fd, pbuf) {
  try {
    var type = 1 == fd || 2 == fd ? 2 : abort();
    HEAP8[pbuf >> 0] = type;
    return 0;
  } catch (e) {
    "undefined" != typeof FS && e instanceof FS.ErrnoError || abort(e);
    return e.errno;
  }
}
```

without FS support, that `_fd_fdstat_get` function is the same than:

```js
function _fd_fdstat_get(fd, pbuf) {
  try {
    var type = 1 == fd || 2 == fd ? 2 : abort();
    HEAP8[pbuf >> 0] = type;
    return 0;
  } catch (e) {
    abort(e);
    return e.errno;
  }
}
```

and the `abort(e)` is the same than `throw e`, hence the function will behave in the exact same way if it's written as:

```js
function _fd_fdstat_get(fd, pbuf) {
    var type = 1 == fd || 2 == fd ? 2 : abort();
    HEAP8[pbuf >> 0] = type;
    return 0;
}
```

I.e. the exceptions will propagate through either way.

Reply:
Oh right, good point. Thanks for the walkthrough, that illustrates the scenario well.

Is there anything specific to `MINIMAL_RUNTIME` here? It looks like this could also be enabled in regular runtime?

Reply:
Yeah, you're right, i've removed the `MINIMAL_RUNTIME` flag.